### PR TITLE
fix 'Use 'for..of' loop for non-array classes' (close #1158)

### LINKED ISSUE #1158: Use 'for..of' loop for non-array classes (NodeList, HTMLCollection and etc.)
- need to patch `babel-preset-es2015-hammerhead` and setup `loose=false` option for `babel-plugin-transform-es2015-for-of`
- rewrite code using `for..of` where it is possible

### PR DESCRIPTION
**Pull request by miherlosev**
@churkin @LavrovArtem

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/event/drag-and-drop/data-transfer-item-list.js [hunk 1]
```diff
@@ -13,9 +13,9 @@ function parseTextUriList (textUriList) {
 
     textUriList = textUriList.split(/\r\n/);
 
-    for (let i = 0; i < textUriList.length; i++) {
-        if (textUriList[0] !== '#')
```
Comment (LavrovArtem) [inline src/client/sandbox/event/drag-and-drop/data-transfer-item-list.js]: Strange behavior. The same condition is repeated each iteration.

Comment (miherlosev) [inline src/client/sandbox/event/drag-and-drop/data-transfer-item-list.js]: Ok. I will investigate this

Comment (miherlosev) [inline src/client/sandbox/event/drag-and-drop/data-transfer-item-list.js]: I propose to apply this small fix.
Also, I've created the separate issue for uri-list conversion #1248.

Note, I've checked uri-list conversion in IE, Firefox and Chrome and it work nowhere.
There is the bug for this case - https://bugs.chromium.org/p/chromium/issues/detail?id=239745

**Comment by miherlosev** [comment]
FPR. see https://github.com/DevExpress/testcafe-hammerhead/pull/1240#discussion_r129802704

**Pull request merged**